### PR TITLE
Fetch go-java-launcher from released distribution instead of checked-in binary

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy [hunk 1]
```diff
@@ -31,18 +32,23 @@ class JavaDistributionPlugin implements Plugin<Project> {
         // force application of java
         project.plugins.apply('java')
 
+        project.configurations.create("goJavaLauncherBinaries")
+        project.dependencies {
+            goJavaLauncherBinaries "com.palantir.launching:go-java-launcher:1.0.0"
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: is it worth allowing this to be overriden?

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy [hunk 2]
```diff
@@ -31,18 +32,23 @@ class JavaDistributionPlugin implements Plugin<Project> {
         // force application of java
         project.plugins.apply('java')
 
+        project.configurations.create("goJavaLauncherBinaries")
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: single quotes, please

**Comment by markelliot** [comment]
code looks okay but your tests are dying

File: src/main/resources/init.sh [hunk 3]
```diff
@@ -27,10 +27,10 @@ pushd "`dirname \"$0\"`/../.." > /dev/null
 # Select launcher binary for this OS
 case "`uname`" in
   Linux*)
-    LAUNCHER_CMD=service/bin/javalauncher-linux-amd64
+    LAUNCHER_CMD=service/bin/linux-amd64/go-java-launcher
     ;;
   Darwin*)
-    LAUNCHER_CMD=service/bin/javalauncher-darwin-amd64
+    LAUNCHER_CMD=service/bin/darwin-amd64/go-java-launcher
```
Comment (markelliot) [inline src/main/resources/init.sh:33]: probably have to adjust the README to match this

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy (hunk 1 above)
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: probably not, but the code would be much easier if didn't explicitly allow it.

Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: (using a gradle configuration also gives you caching for free which is quite handy)

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy (hunk 2 above)
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: done

File: src/main/resources/init.sh (hunk 3 above)
Comment (uschi2000) [inline src/main/resources/init.sh:33]: done

**Comment by uschi2000** [comment]
no ideas why tests died, trying again.

**Comment by uschi2000** [comment]
no idea why this is failing, will look tomorrow

**Comment by markelliot** [comment]
One thing we might want to consider here is this plugin bundling the go artifacts into the produced jars rather than requiring every downstream project to add the jcenter repo explicitly.

**Comment by uschi2000** [comment]
yep, could do

**Pull request merged**